### PR TITLE
Add runs search aliases for run name and start time to correspond to UI column names

### PR DESCRIPTION
**Pull request by apurva-koti**
Signed-off-by: Apurva Koti <apurva.koti@databricks.com>

## Related Issues/PRs

 #xxx

## What changes are proposed in this pull request?

Add `run_name` search aliases `run name, Run name, Run Name` and `start_time` aliases `created` and `Created`.

## How is this patch tested?

- [ ] Existing unit/integration tests
- [x] New unit/integration tests
- [ ] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

Searching runs by run names now work with the aliases `run name`, `Run name` and `Run Name`. Searching runs by start time now works with the aliases `created` and `Created`.

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [x] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [x] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: docs/source/search-runs.rst [hunk 1]
```diff
@@ -88,6 +88,8 @@ Run Attributes
 
 You can search using the following run attributes contained in :py:class:`mlflow.entities.RunInfo`: ``run_id``, ``run_name``, ``status``, ``artifact_uri``, ``user_id``, ``start_time`` and ``end_time``. The ``run_id``, ``run_name``, ``status``, ``user_id`` and ``artifact_uri`` attributes have string values, while ``start_time`` and ``end_time`` are numeric. Other fields in ``mlflow.entities.RunInfo`` are not searchable.
 
+``Run name``, ``Run Name`` and ``run name`` are aliases for ``run_name``. ``created`` and ``Created`` are aliases for ``start_time``.
```
Comment (dbczumar) [inline docs/source/search-runs.rst:91]: Can we add examples for each of these?

File: tests/store/tracking/test_sqlalchemy_store.py [hunk 2]
```diff
@@ -2166,6 +2184,38 @@ def test_search_runs_run_id(self):
...
+        )
+        assert [r.info.run_id for r in result] == [run_id1, run_id2]
+
+        result = self.store.search_runs(
+            [exp_id],
+            filter_string="attributes.run_name = 'name'",
+            run_view_type=ViewType.ACTIVE_ONLY,
+            order_by=["attributes.Created DESC"],
+        )
+        assert [r.info.run_id for r in result] == [run_id2, run_id1]
```
Comment (dbczumar) [inline tests/store/tracking/test_sqlalchemy_store.py:2217]: all of these filter strings appear to be the same. Can we make them different? Also, can we add tests for "Created"?

File: tests/store/tracking/test_file_store.py [hunk 3]
```diff
@@ -1227,6 +1248,39 @@ def test_search_runs_run_id(self):
...
+            order_by=["attributes.created ASC"],
+        )
+        assert [r.info.run_id for r in result] == [run_id1, run_id2]
+
+        result = fs.search_runs(
+            [exp_id],
+            filter_string="attributes.run_name = 'name'",
+            run_view_type=ViewType.ACTIVE_ONLY,
+            order_by=["attributes.Created DESC"],
+        )
```
Comment (dbczumar) [inline tests/store/tracking/test_file_store.py:1281]: all of these filter strings appear to be the same. Can we make them different? Also, can we add tests for "Created" in the filter?

File: tests/store/tracking/test_sqlalchemy_store.py (hunk 2 above)
Comment (apurva-koti) [reply tests/store/tracking/test_sqlalchemy_store.py:2217]: Created is being tested here in the order by, not the filter (hence the filter string is the same)

File: tests/store/tracking/test_file_store.py (hunk 3 above)
Comment (apurva-koti) [reply tests/store/tracking/test_file_store.py:1281]: https://github.com/mlflow/mlflow/pull/7492#discussion_r1044896778

Comment (dbczumar) [reply tests/store/tracking/test_file_store.py:1281]: Then the title of the test case seems to be misleading

File: tests/store/tracking/test_sqlalchemy_store.py (hunk 2 above)
Comment (dbczumar) [reply tests/store/tracking/test_sqlalchemy_store.py:2217]: Then the title of the test case seems to be misleading

File: tests/store/tracking/test_file_store.py (hunk 3 above)
Comment (dbczumar) [reply tests/store/tracking/test_file_store.py:1281]: I see - can we test filtering on Created too?

File: tests/store/tracking/test_sqlalchemy_store.py (hunk 2 above)
Comment (dbczumar) [reply tests/store/tracking/test_sqlalchemy_store.py:2217]: I see - can we test filtering on Created too?

Comment (apurva-koti) [reply tests/store/tracking/test_sqlalchemy_store.py:2217]: Sure!

File: docs/source/search-runs.rst [hunk 4]
```diff
@@ -35,13 +35,20 @@ Example Expressions
...
     attributes.status = "FINISHED"
 
-
 - Search for all failed runs.
 
   .. code-block:: sql
 
     attributes.status = "FAILED"
 
+- Search for runs created after UNIX timestamp ``1640502832``.
```
Comment (dbczumar) [inline docs/source/search-runs.rst:44]: Nit: I think we use millisecond resolution for UNIX timestamps. This appears to be in second resolution. Can we update this?

**Review by dbczumar — APPROVED** [review]
LGTM once https://github.com/mlflow/mlflow/pull/7492/files#r1044911729 is addressed. Thanks a bunch, @apurva-koti !

File: docs/source/search-runs.rst (hunk 4 above)
Comment (apurva-koti) [reply docs/source/search-runs.rst:44]: Yikes 😬

File: tests/store/tracking/test_sqlalchemy_store.py [hunk 5]
```diff
@@ -2166,6 +2184,59 @@ def test_search_runs_run_id(self):
...
+            run_view_type=ViewType.ACTIVE_ONLY,
+        )
+        assert set([r.info.run_id for r in result]) == {run_id2}
+
+        result = self.store.search_runs(
+            [exp_id],
+            filter_string="attributes.Created > 2",
+            run_view_type=ViewType.ACTIVE_ONLY,
+        )
+        assert set([r.info.run_id for r in result]) == set([])
```
Comment (harupy) [inline tests/store/tracking/test_sqlalchemy_store.py:2238]:
```suggestion
        assert result == []
```

File: tests/store/tracking/test_file_store.py [hunk 6]
```diff
@@ -1227,6 +1248,60 @@ def test_search_runs_run_id(self):
...
+            run_view_type=ViewType.ACTIVE_ONLY,
+        )
+        assert set([r.info.run_id for r in result]) == {run_id2}
+
+        result = fs.search_runs(
+            [exp_id],
+            filter_string="attributes.Created > 2",
+            run_view_type=ViewType.ACTIVE_ONLY,
+        )
+        assert set([r.info.run_id for r in result]) == set([])
```
Comment (harupy) [inline tests/store/tracking/test_file_store.py:1303]:
```suggestion
assert result == []
```

File: tests/store/tracking/test_file_store.py [hunk 7]
```diff
@@ -1227,6 +1248,60 @@ def test_search_runs_run_id(self):
...
+            order_by=["attributes.Created DESC"],
+        )
+        assert [r.info.run_id for r in result] == [run_id2, run_id1]
+
+        result = fs.search_runs(
+            [exp_id],
+            filter_string="attributes.start_time > 0",
+            run_view_type=ViewType.ACTIVE_ONLY,
+        )
+        assert set([r.info.run_id for r in result]) == {run_id1, run_id2}
```
Comment (harupy) [inline tests/store/tracking/test_file_store.py:1289]:
```suggestion
        assert {r.info.run_id for r in result]} == {run_id1, run_id2}
```

**Review by harupy — APPROVED** [review]
LGTM once the lint errors are fixed!

File: tests/store/tracking/test_sqlalchemy_store.py [hunk 8]
```diff
@@ -2166,6 +2184,59 @@ def test_search_runs_run_id(self):
...
+            run_view_type=ViewType.ACTIVE_ONLY,
+        )
+        assert set([r.info.run_id for r in result]) == {run_id1, run_id2}
+
+        result = self.store.search_runs(
+            [exp_id],
+            filter_string="attributes.created > 1",
+            run_view_type=ViewType.ACTIVE_ONLY,
+        )
+        assert set([r.info.run_id for r in result]) == {run_id2}
```
Comment (harupy) [inline tests/store/tracking/test_sqlalchemy_store.py:2231]:
```suggestion
        assert [r.info.run_id for r in result] == [run_id2]
```

File: tests/store/tracking/test_sqlalchemy_store.py [hunk 9]
```diff
@@ -2166,6 +2184,59 @@ def test_search_runs_run_id(self):
...
+            order_by=["attributes.Created DESC"],
+        )
+        assert [r.info.run_id for r in result] == [run_id2, run_id1]
+
+        result = self.store.search_runs(
+            [exp_id],
+            filter_string="attributes.start_time > 0",
+            run_view_type=ViewType.ACTIVE_ONLY,
+        )
+        assert set([r.info.run_id for r in result]) == {run_id1, run_id2}
```
Comment (harupy) [inline tests/store/tracking/test_sqlalchemy_store.py:2224]:
```suggestion
        assert {r.info.run_id for r in result} == {run_id1, run_id2}
```

or 

```suggestion
        assert [r.info.run_id for r in result] == [run_id1, run_id2]
```

if the run order is deterministic.

**Pull request merged**